network/src/noise/handshake.rs

```diff
@@ -429,31 +385,23 @@ impl NoiseUpgrader {
         if let Some(anti_replay_timestamps) = self.auth_mode.anti_replay_timestamps() {
             // check that the payload received as the client timestamp (in seconds)
             if payload.len() != AntiReplayTimestamps::TIMESTAMP_SIZE {
-                // TODO: security logging (mimoo)
-                return Err(io::Error::new(
-                    io::ErrorKind::InvalidData,
-                    "noise: client initiated connection without an 8-byte timestamp",
+                return Err(NoiseHandshakeError::MissingAntiReplayTimestamp(
+                    remote_peer_short,
                 ));
             }
+
             let mut client_timestamp = [0u8; AntiReplayTimestamps::TIMESTAMP_SIZE];
             client_timestamp.copy_from_slice(&payload);
             let client_timestamp = u64::from_le_bytes(client_timestamp);
 
             // check the timestamp is not a replay
             let mut anti_replay_timestamps = anti_replay_timestamps.write().map_err(|_| {
-                io::Error::new(
-                    io::ErrorKind::Other,
-                    "noise: unable to read anti_replay_timestamps lock",
-                )
+                NoiseHandshakeError::LockPoisoned(remote_peer_short, "anti_replay_timestamps")
```

Comment:
Sorry, I'm making you iterate on this, but let's just do `expect` for poisoned locks, otherwise we have to take manual intervention when it dies.

Reply:
Poisoned locked? IMO Noise shouldn't even produce an Error like this and should just `unwrap` it

Reply:
`expect` or `unwrap` works, something where we need to restart anyways

Reply:
This is a case where `unwrap` is probably preferred, just so that you don't have to provide a generic "poisoned lock" message which doesn't add any value.